### PR TITLE
Submit QIR Job From Qiskit to IonQ

### PR DESCRIPTION
**Pull request by ScottCarda-MS**
Allows for the submission of QIR jobs from Qiskit python to IonQ targets.

### CODE REVIEW

**Review by fedonman — APPROVED** [review]

File: azure-quantum/azure/quantum/target/microsoft/qio/simulated_annealing.py [hunk 1]
```diff
@@ -16,10 +16,10 @@
 
 
 class SimulatedAnnealing(Solver):
-    target_names = [
+    target_names = (
```
Comment (msoeken) [inline azure-quantum/azure/quantum/target/microsoft/qio/simulated_annealing.py:19]: Is this change intended? Just asking because the code is out of the scope for the PR tilte.

**Review by msoeken — APPROVED** [review]

File: azure-quantum/azure/quantum/target/microsoft/qio/simulated_annealing.py (hunk 1 above)
Comment (ScottCarda-MS) [reply azure-quantum/azure/quantum/target/microsoft/qio/simulated_annealing.py:19]: It is not related. I was fixing a bug where a target_name was just a single element and not being treated as a list, and saw that this was the only target_names that was a list instead of a tuple. I see that the bug was independently fixed while I was working on this in the same way.

File: azure-quantum/azure/quantum/qiskit/backends/ionq.py [hunk 2]
```diff
@@ -53,7 +53,9 @@ def _prepare_job_metadata(self, circuit, **kwargs):
 
     def _translate_input(self, circuit, data_format, input_params, to_qir_kwargs={}):
         """ Translates the input values to the format expected by the AzureBackend. """
-        if data_format == "ionq.circuit.v1":
+        if "targetCapability" in input_params and input_params["targetCapability"] == "BasicExecution":
```
Comment (DmitryVasilevsky) [inline azure-quantum/azure/quantum/qiskit/backends/ionq.py:56]: So targetCapability can be "ionq" (line 34) or "BasicExecution"? Or these are different target capabilities?

Comment (ScottCarda-MS) [reply azure-quantum/azure/quantum/qiskit/backends/ionq.py:56]: Yes that's right. I didn't know what to use for the default target capability for ionq, but I saw that quantuniuum has default capability of "openqasm", so I went with "ionq" for the default for ionq. It isn't a capability that would be recognized, I don't imagine, but the point is for there to be a default option other than "BasicExecution".

If there is a more correct way to set this up, please let me know, because I just took a guess here.

File: azure-quantum/azure/quantum/qiskit/backends/ionq.py [hunk 3]
```diff
@@ -31,7 +31,7 @@ class IonQBackend(AzureBackend):
 
     @classmethod
     def _default_options(cls):
-        return Options(shots=500)
+        return Options(shots=500, targetCapability="ionq")
```
Comment (cesarzc) [inline azure-quantum/azure/quantum/qiskit/backends/ionq.py:34]: Is there a reason for specifying the `targetCapability` by default? I believe the previous behavior should still be the default.

**Comment by cesarzc** [comment]
@anpaz, could you please take a look? I did an initial review but since you are more familiar with the code base it would be good if you could also review it.

File: azure-quantum/azure/quantum/qiskit/backends/ionq.py (hunk 3 above)
Comment (ScottCarda-MS) [reply azure-quantum/azure/quantum/qiskit/backends/ionq.py]: The behavior is the same by default. Only if someone sets targetCapability to "BasicExecution" is the qir translation triggered.

Comment (ScottCarda-MS) [reply azure-quantum/azure/quantum/qiskit/backends/ionq.py]: I've removed the default targetCapability for ionq since it was causing confusion.

File: azure-quantum/azure/quantum/qiskit/backends/ionq.py (hunk 2 above)
Comment (ScottCarda-MS) [reply azure-quantum/azure/quantum/qiskit/backends/ionq.py:56]: I've removed the default targetCapability.
